system/core/database.php

```diff
@@ -559,8 +559,52 @@ function serveFile($filename, $root) {
 			http_response_code(404);
 			die("Invalid filename: $filename");
 		}
+		// TODO: Use X-Sendfile header to let Apache serve the file?
+		$filesize = filesize($filename);
+		$headers = getallheaders();
+		$matches = [];
+		// Only supports fully-qualified bytes=start-end ranges,
+		// not `bytes=-suffixlength` or `bytes=start-` or
+		// multi-part ranges.
+		if (array_key_exists('Range', $headers) && preg_match('/bytes=(\d+)-(\d+)$/', $headers['Range'], $matches)) {
```

Comment:
should it give an error if a multi part is requested but could not be returned?

Reply:
I'm not sure what error would be appropriate here. If we just return the entire file, the client could see the absence of the range headers and conclude that it wasn't supported, I hope...

Reply:
Seems ignoring the header is always allowed: https://stackoverflow.com/a/6204397/740048